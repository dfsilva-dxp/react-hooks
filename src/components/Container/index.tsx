import styled, { css } from "styled-components";

const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin: 0 auto;
    padding: 0 ${theme.spacings.xsmall};

    @media (min-width: 576px) {
      max-width: ${theme.grid.container.sm};
    }
    @media (min-width: 768px) {
      max-width: ${theme.grid.container.md};
    }
    @media (min-width: 992px) {
      max-width: ${theme.grid.container.lg};
    }
    @media (min-width: 1200px) {
      max-width: ${theme.grid.container.xlg};
    }
    @media (min-width: 1400px) {
      max-width: ${theme.grid.container.xxlg};
    }
  `}
`;

export default Container;
