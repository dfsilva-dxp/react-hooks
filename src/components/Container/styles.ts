import styled, { css } from "styled-components";

export const Content = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.grid.container};
    margin: 0 auto;
    padding: 0 ${theme.spacings.xsmall};
  `}
`;
