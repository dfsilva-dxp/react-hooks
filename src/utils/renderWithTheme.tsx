import { ReactNode } from "react";
import { render, RenderResult } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { MyTheme } from "../styles";

export const renderWithTheme = (children: ReactNode): RenderResult =>
  render(<ThemeProvider theme={MyTheme}>{children}</ThemeProvider>);
