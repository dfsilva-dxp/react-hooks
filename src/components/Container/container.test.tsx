import { screen } from "@testing-library/react";

import Container from ".";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

describe("<Container/>", () => {
  it("should render with width: 100% in small size", () => {
    const { container } = renderWithTheme(
      <Container>
        <h1>React hooks</h1>
      </Container>
    );

    const heading = screen.getByRole("heading", { name: /react hooks/i });
    const parent = heading.parentElement;

    expect(parent).toHaveStyle({
      width: "100%",
      margin: "0 auto",
      padding: "0 1rem"
    });

    expect(container).toMatchSnapshot();
  });

  it("should render with a width 33.75rem when the screen has a min-width 576px", () => {
    renderWithTheme(
      <Container>
        <h1>React hooks</h1>
      </Container>
    );

    const heading = screen.getByRole("heading", { name: /react hooks/i });
    const parent = heading.parentElement;

    expect(parent).toHaveStyleRule("max-width", "33.75rem", {
      media: "(min-width: 576px)"
    });
  });

  it("should render with a width 45rem when the screen has a min-width 768px", () => {
    renderWithTheme(
      <Container>
        <h1>React hooks</h1>
      </Container>
    );

    const heading = screen.getByRole("heading", { name: /react hooks/i });
    const parent = heading.parentElement;

    expect(parent).toHaveStyleRule("max-width", "45rem", {
      media: "(min-width: 768px)"
    });
  });

  it("should render with a width 60rem when the screen has a min-width 992px", () => {
    renderWithTheme(
      <Container>
        <h1>React hooks</h1>
      </Container>
    );

    const heading = screen.getByRole("heading", { name: /react hooks/i });
    const parent = heading.parentElement;

    expect(parent).toHaveStyleRule("max-width", "60rem", {
      media: "(min-width: 992px)"
    });
  });

  it("should render with a width 71.25rem when the screen has a min-width 1200px", () => {
    renderWithTheme(
      <Container>
        <h1>React hooks</h1>
      </Container>
    );

    const heading = screen.getByRole("heading", { name: /react hooks/i });
    const parent = heading.parentElement;

    expect(parent).toHaveStyleRule("max-width", "71.25rem", {
      media: "(min-width: 1200px)"
    });
  });

  it("should render with a width 82.5rem when the screen has a min-width 1400px", () => {
    renderWithTheme(
      <Container>
        <h1>React hooks</h1>
      </Container>
    );

    const heading = screen.getByRole("heading", { name: /react hooks/i });
    const parent = heading.parentElement;

    expect(parent).toHaveStyleRule("max-width", "82.5rem", {
      media: "(min-width: 1400px)"
    });
  });
});
