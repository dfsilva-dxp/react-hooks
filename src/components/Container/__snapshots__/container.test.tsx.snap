// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container/> should render with width: 100% in small size 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  padding: 0 1rem;
}

@media (min-width:576px) {
  .c0 {
    max-width: 33.75rem;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 45rem;
  }
}

@media (min-width:992px) {
  .c0 {
    max-width: 60rem;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 71.25rem;
  }
}

@media (min-width:1400px) {
  .c0 {
    max-width: 82.5rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h1>
      React hooks
    </h1>
  </div>
</div>
`;
